Bluetooth: Mesh: Doc: Clarify settings cache configuration (#14722)
nrfconnect/sdk-nrf: doc/nrf/protocols/bt/bt_mesh/configuring.rst

This can for example happen when storing a large size replay protection list.
It is recommended to configure the settings subsystem's internal caches to improve the performance.

The Settings NVS name cache reduces the number of search loops of internal parameter identifiers, keeping them in memory.

* :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE`.

The size of the Settings NVS name cache, :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE_SIZE`, is recommended to be at least equal to the number of settings entries the device is expected to store.

> The Mesh stack stores the following data persistently:

Reviewer: ```suggestion
The Bluetooth Mesh stack stores the following data persistently:
```

The size of the Settings NVS name cache, :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE_SIZE`, is recommended to be at least equal to the number of settings entries the device is expected to store.

The Bluetooth Mesh stack stores the following data persistently:

* Network information (primary address and device key)

> * Configuration parameters (supported features, default TTL, network transmit- and relay retransmit parameters)

Reviewer: Why hyphen in network transmit-?

Author: The hyphen was included to not write "parameters" twice. I. e. instead of `network transmit parameters and relay retransmit parameters`

Author: The same goes for the other cases regarding "models". Do you still think I should remove the hyphen?

Reviewer: Yeah, I think we can remove these :) You would usually use it with compounds or hyphenated terms that are listed, not necessary here. 
...network transmit and relay retransmit parameters, is ok without the hyphen.
The same goes for the models.

Author: Removed hyphens and added changelog entry :)

* Network key(s) (the amount of entries is controlled by :kconfig:option:`CONFIG_BT_MESH_SUBNET_COUNT`)
* Label UUIDs for virtual addressing (the amount of entries is controlled by :kconfig:option:`CONFIG_BT_MESH_LABEL_COUNT`)
* RPL entries (the RPL size is controlled by :kconfig:option:`CONFIG_BT_MESH_CRPL`)

> The following data is stored by the Mesh stack per model:

Reviewer: ```suggestion
The following data is stored for each model by the Bluetooth Mesh stack:
```

* Model subscription state
* Model publication state
* Bound application key(s)
* Subscription list for group addresses
* Subscription list for virtual addresses
* Label UUIDs the model is subscribed to

> * Model specific data

Reviewer: ```suggestion
* Model-specific data
```

>   * Model data stored persistently can be found under the ``Persistent storage`` section of the corresponding model documentation.

Reviewer: ```suggestion
Model data stored persistently can be found under the ``Persistent storage`` section of the corresponding model documentation.
```

* Label UUIDs the model is subscribed to
* Model-specific data

Model data stored persistently can be found under the ``Persistent storage`` section of the corresponding model documentation.

Using the :ref:`bluetooth_mesh_sensor_server` sample as an example, configured according to the sample's :ref:`configuration guide <bluetooth_mesh_sensor_server_conf_models>`, results in the following list of possible entries (entries mentioned above are not included unless specifying the amount of entries):

* 32 RPL entries - since the default Networked Lighting Control (NLC) configuration is used (:kconfig:option:`CONFIG_BT_MESH_NLC_PERF_DEFAULT` is set), the RPL size is 32.
* Application keys - three keys are used.

> * Bound application keys - each of the three Sensor Server- and Sensor Setup Server models has one bound application key.

Reviewer: ```suggestion
* Bound application keys - each of the three Sensor Server and Sensor Setup Server models has one bound application key.
```

> * Virtual addressing is not used

Reviewer: ```suggestion
* Virtual addressing is not used.
```

* Network keys - only one key is used.

> * Model subscriptions - each of the three Sensor Server- and Sensor Setup Server models subscribes to a group address.

Reviewer: ```suggestion
* Model subscriptions - each of the three Sensor Server and Sensor Setup Server models subscribes to a group address.
```

> * :ref:`Sensor Server model data <bt_mesh_sensor_srv_persistent_readme>` - each of the Three Sensor Server models stores the following data:

Reviewer: ```suggestion
* :ref:`Sensor Server model data <bt_mesh_sensor_srv_persistent_readme>` - each of the three Sensor Server models stores the following data:
```

  * Minimum interval
  * Delta thresholds
  * Fast period divisor
  * Fast cadence range

* The following values are stored in the sample:

  * Temperature range
  * Presence motion threshold
  * Ambient light level gain

Adding up all entries, it is worth setting the cache size to minimum 71.

NVS lookup cache reduces the number of search loops within NVS' application table.

> * :kconfig:option:`CONFIG_NVS_LOOKUP_CACHE`.

Reviewer: Consider moving this up to before line 193.

This can for example happen when storing a large size replay protection list.
It is recommended to configure the settings subsystem's internal caches to improve the performance.

The Settings NVS name cache reduces the number of search loops of internal parameter identifiers, keeping them in memory.

* :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE`

> The size of the Settings NVS name cache, :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE_SIZE`, is recommended to be at least equal to the number of setting entries the device is expected to store. These entries may include entries in the RPL, whose size is given by :kconfig:option:`CONFIG_BT_MESH_CRPL`, :ref:`configuration values <ug_bt_mesh_configuring_optional_features>`, application- and network keys, application specific states etc. Using :ref:`bluetooth_mesh_sensor_server` as an example configured according to the sample's :ref:`configuration guide <bluetooth_mesh_sensor_server_conf_models>`, this has the following possible entries:

Reviewer: ```suggestion
The size of the Settings NVS name cache, :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE_SIZE`, is recommended to be at least equal to the number of settings entries the device is expected to store.
These entries may include entries in the RPL, whose size is given by :kconfig:option:`CONFIG_BT_MESH_CRPL`, :ref:`configuration values <ug_bt_mesh_configuring_optional_features>`, application- and network keys, application-specific states, and the like.
Using the :ref:`bluetooth_mesh_sensor_server` sample as an example, configured according to the sample's :ref:`configuration guide <bluetooth_mesh_sensor_server_conf_models>`, results in the following list of possible entries:
```

> The size of the Settings NVS name cache, :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE_SIZE`, is recommended to be at least equal to the number of setting entries the device is expected to store. These entries may include entries in the RPL, whose size is given by :kconfig:option:`CONFIG_BT_MESH_CRPL`, :ref:`configuration values <ug_bt_mesh_configuring_optional_features>`, application- and network keys, application specific states etc. Using :ref:`bluetooth_mesh_sensor_server` as an example configured according to the sample's :ref:`configuration guide <bluetooth_mesh_sensor_server_conf_models>`, this has the following possible entries:

Reviewer 2: I'd suggest to have a generic section. Most of the list items starting from line 201 could be moved into that section. It could say _something like_:
```
The mesh stack stores the following data persistently (each bullet-point is a separate setting entry):
- Network information (primary address and device key)
- Configuration parameters (features support, default TTL, Network Transmit and Relay Retransmit states)
- IV index
- Sequence number
- Heartbeat publication state
- Application key (see :kconfig:option:`CONFIG_BT_MESH_APP_KEY_COUNT`)
- Network key (see :kconfig:option:`CONFIG_BT_MESH_SUBNET_COUNT`)
- Label UUIDs for virtual addressing (see :kconfig:option:`BT_MESH_LABEL_COUNT`)
- RPL entry (see :kconfig:option:`CONFIG_BT_MESH_CRPL`)

The following data is stored by the mesh stack per each model (each bullet-point is a separate setting entry):
- Model subscription state
- Model publication state
- Bound application keys
- Subscription list (group and virtual addresses)
- Label UUIDs the model is subscribed to
- Model specific data
  - The information about a model specific data stored persistently can be found under the ``Persistent storage`` section of corresponding model documentation.
```

And then we can provide example:
```
Using :ref:`bluetooth_mesh_sensor_server` as an example, this has the following possible entries:
- 32 RPL entries
- 3 application keys
...
etc.
```
\+ mention the model specific data and the sample specific data (btw, we have `Persistent storage` section in each model saying what each models store, e.g.: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/bluetooth_services/mesh/sensor_srv.html#persistent-storage, so we can refer to in in the example).

Author: Added generic section. In the example I only mention the generics which possibly has variable amount of entries

> * RPL entries. Since the default Networked Lighting Control (NLC) config :kconfig:option:`CONFIG_BT_MESH_NLC_PERF_DEFAULT` is used, the RPL size is 32.
> * Application key(s). Assume three are used.
> * Bound application keys. Each of the three Sensor Servers and Sensor Setup Servers has one bound application key.
> * Network key(s). Assume only one key is used.
> * Model subscription(s). Each of the three Sensor Servers and Sensor Setup Servers models subscribes to a group address.
> * Model publication information. Each of the three Sensor Server models publishes to a group address.
> * Virtual address(es). Assume these are not used.
> * Label UUIDs for virtual addressing. Assume these are not used.
> * Configuration values
> * Heartbeat publication information
> * IV index
> * Network information
> * Sequence number
> * Model data. Each of the Three Sensor Server models stores the following data:

Reviewer: ```suggestion
* RPL entries - Since the default Networked Lighting Control (NLC) configuration is used (:kconfig:option:`CONFIG_BT_MESH_NLC_PERF_DEFAULT` is set), the RPL size is 32.
* Application key(s) - Assuming three keys are used.
* Bound application keys - Each of the three Sensor Server and Sensor Setup Server models has one bound application key.
* Network key(s) - Assuming only one key is used.
* Model subscription(s) - Each of the three Sensor Server and Sensor Setup Server models subscribes to a group address.
* Model publication information - Each of the three Sensor Server models publishes to a group address.
* Virtual address(es) - Assuming these are not used.
* Label UUIDs for virtual addressing - Assuming these are not used.
* Configuration values
* Heartbeat publication information
* IV index
* Network information
* Sequence number
* Model data - Each of the three Sensor Server models stores the following data:
```

  * Minimum interval
  * Delta thresholds
  * Fast period divisor
  * Fast cadence range

* The following values are stored in the sample:

  * Temperature range
  * Presence sensor motion threshold

>   * Ambient light level gain

Reviewer 2: Btw, the sample documentation is wrong because it doesn't say about the ambient light level gain and the presence threshold: https://github.com/nrfconnect/sdk-nrf/blob/e661da007b6d82ea7be70f10e92d8ffc9e8c3af1/samples/bluetooth/mesh/sensor_server/README.rst?plain=1#L116

Could you fix it as well?

Author: Added separate commit for this
